Check flake8 errors:

```
./addons/account_voucher/migrations/9.0.1.0/post-migration.py:27:1: E302 expected 2 blank lines, found 1
./addons/account_voucher/migrations/9.0.1.0/post-migration.py:37:1: E302 expected 2 blank lines, found 1
./addons/account_voucher/migrations/9.0.1.0/post-migration.py:88:9: E128 continuation line under-indented for visual indent
```

and can you please add some tests?

--- a/addons/account_voucher/migrations/9.0.1.0/post-migration.py
+++ b/addons/account_voucher/migrations/9.0.1.0/post-migration.py
@@ -4,44 +4,10 @@
 from openupgradelib import openupgrade
 
 
-def get_ref(cr, module, xml_id):
-    """Get reference (res_id) from module and xml_id from ir_model_data."""
-    cr.execute("""
-        SELECT res_id
-        FROM ir_model_data
-        WHERE module = %s AND name = %s
-    """, (module, xml_id))
-    return cr.fetchone()[0]
-
-
-def migrate_voucher_types(cr):
-    """Migrate voucher types."""
-    # TODO: Should we really keep receipt and payment records? Might contain
-    # duplicate info with the sales and purchases for which they have been
-    # made, especially now that we migrate to account.payment.
-    openupgrade.map_values(
-        cr, openupgrade.get_legacy_name('type'), 'voucher_type',
-        [('receipt', 'sale'), ('payment', 'purchase')],
-        table='account_voucher')
-
-def create_voucher_line_tax_lines(cr):
-    """Migrate tax information on voucher lines to m2m relation."""
-    cr.execute(
-        "insert into account_tax_account_voucher_line_rel "
-        "(account_tax_id, account_voucher_line_id) "
-        "select v.tax_id, l.id "
-        "from account_voucher v "
-        "join account_voucher_line l on l.voucher_id=v.id "
-        "where v.tax_id is not null")
-
-def create_payments_from_vouchers(cr):
+def create_payments_from_vouchers(env):
     """Create account.payment rows from appropiate account.voucher rows."""
-    receipt_method = get_ref(
-        cr, 'account', 'account_payment_method_manual_in'
-    )
-    payment_method = get_ref(
-        cr, 'account', 'account_payment_method_manual_out'
-    )
+    receipt_method = env.ref('account.account_payment_method_manual_in')
+    payment_method = env.ref('account.account_payment_method_manual_out')
     # Keep original id, to later migrate references from other tables,
     # although it seems not te be needed for now.
     # Possible states of account.payment: draft, posted, sent, reconciled
@@ -49,9 +15,10 @@ def create_payments_from_vouchers(cr):
     # "type" field in 8.0 renamed to "voucher_type" during migration.
     # Field destination_journal_id is used for internal transfer payments.
     # Not used here.
-    # As far as I can see there is no connection between invoices or moves
-    # in 8.0 taht we can or have to migrate to 9.0.
-    cr.execute("""
+    # As far as I can see there is no connection between invoices in 8.0
+    # that we can or have to migrate to 9.0.
+    env.cr.execute(
+        """\
         INSERT INTO account_payment (
             id, create_date, communication, company_id, currency_id,
             partner_id, partner_type,
@@ -61,45 +28,87 @@ def create_payments_from_vouchers(cr):
             name, amount, payment_type, payment_reference
         )
         SELECT
-            v.id, v.create_date, v.comment, v.company_id,
-            v.payment_rate_currency_id, v.partner_id,
+            av.id, av.create_date, av.comment, av.company_id,
+            av.payment_rate_currency_id, av.partner_id,
             CASE
-               WHEN v.voucher_type = 'receipt' THEN 'customer'
+               WHEN av.voucher_type = 'receipt' THEN 'customer'
                ELSE 'supplier'
             END,
-            CASE WHEN v.voucher_type = 'receipt' THEN %s ELSE %s END,
-            v.create_date,
+            CASE WHEN av.voucher_type = 'receipt' THEN %s ELSE %s END,
+            av.create_date,
             CASE
-                WHEN v.payment_option = 'with_writeoff' THEN 'reconcile'
+                WHEN av.payment_option = 'with_writeoff' THEN 'reconcile'
                 ELSE 'open'
             END,
-            v.journal_id, v.state, v.writeoff_acc_id, v.create_uid,
-            v.write_date, v.write_uid,
-            COALESCE(v.name, v.comment),
-            v.amount,
+            av.journal_id, av.state, av.writeoff_acc_id, av.create_uid,
+            av.write_date, av.write_uid,
+            COALESCE(av.name, av.comment),
+            av.amount,
             CASE
-               WHEN v.voucher_type = 'receipt' THEN 'inbound'
+               WHEN av.voucher_type = 'receipt' THEN 'inbound'
                ELSE 'outbound'
             END,
-            v.reference
-        FROM account_voucher v
-        WHERE v.voucher_type IN ('receipt', 'payment')
-        AND v.state in ('draft', 'posted')""",
-        (receipt_method, payment_method)
+            av.reference
+        FROM account_voucher av
+        WHERE av.voucher_type IN ('receipt', 'payment')
+        AND av.state in ('draft', 'posted')
+        """,
+        (receipt_method.id, payment_method.id)
     )
-    cr.execute(
+    env.cr.execute(
         """SELECT COALESCE(MAX(id), 0) AS max_id FROM account_payment"""
     )
-    max_id = cr.fetchone()[0] + 1
-    cr.execute(
+    max_id = env.cr.fetchone()[0] + 1
+    env.cr.execute(
         """ALTER SEQUENCE account_payment_id_seq RESTART WITH %s""",
         (max_id,)
     )
+    # Statement below works because new payments have same id as old vouchers
+    env.cr.execute(
+        """\
+        UPDATE account_move_line aml
+        SET payment_id = (
+            SELECT av.id
+            FROM account_voucher_line avl
+            JOIN account_voucher av ON av.id = avl.voucher_id
+            WHERE avl.move_line_id = aml.id
+        )
+        WHERE aml.id IN (
+            SELECT avl.move_line_id
+            FROM account_voucher_line avl
+            JOIN account_voucher av ON av.id = avl.voucher_id
+            WHERE av.voucher_type IN ('receipt', 'payment')
+              AND NOT avl.move_line_id IS NULL
+              AND NOT av.move_id IS NULL
+        )
+        """
+    )
 
 
-@openupgrade.migrate()
-def migrate(cr, version):
+def delete_payment_vouchers(env):
+    """Delete payment_vouchers. Info is now in account_payment."""
+    env.cr.execute(
+        """\
+        DELETE from account_voucher
+        WHERE voucher_type IN ('receipt', 'payment')
+        """
+    )
+
+
+def create_voucher_line_tax_lines(env):
+    """Migrate tax information on voucher lines to m2m relation."""
+    env.cr.execute(
+        "insert into account_tax_account_voucher_line_rel "
+        "(account_tax_id, account_voucher_line_id) "
+        "select v.tax_id, l.id "
+        "from account_voucher v "
+        "join account_voucher_line l on l.voucher_id=v.id "
+        "where v.tax_id is not null")
+
+
+@openupgrade.migrate(use_env=True)
+def migrate(env, version):
     """Control function for account_voucher migration."""
-    create_payments_from_vouchers(cr)
-    migrate_voucher_types(cr)
-    create_voucher_line_tax_lines(cr)
+    create_payments_from_vouchers(env)
+    delete_payment_vouchers(env)
+    create_voucher_line_tax_lines(env)

--- a/addons/account_voucher/migrations/9.0.1.0/post-migration.py
+++ b/addons/account_voucher/migrations/9.0.1.0/post-migration.py
@@ -4,12 +4,28 @@
 from openupgradelib import openupgrade
 
 
-@openupgrade.migrate()
-def migrate(cr, version):
+def get_ref(cr, module, xml_id):
+    """Get reference (res_id) from module and xml_id from ir_model_data."""
+    cr.execute("""
+        SELECT res_id
+        FROM ir_model_data
+        WHERE module = %s AND name = %s
+    """, (module, xml_id))
+    return cr.fetchone()[0]
+
+
+def migrate_voucher_types(cr):
+    """Migrate voucher types."""
+    # TODO: Should we really keep receipt and payment records? Might contain
+    # duplicate info with the sales and purchases for which they have been
+    # made, especially now that we migrate to account.payment.
     openupgrade.map_values(
         cr, openupgrade.get_legacy_name('type'), 'voucher_type',
         [('receipt', 'sale'), ('payment', 'purchase')],
         table='account_voucher')
+
+def create_voucher_line_tax_lines(cr):
+    """Migrate tax information on voucher lines to m2m relation."""
     cr.execute(
         "insert into account_tax_account_voucher_line_rel "
         "(account_tax_id, account_voucher_line_id) "
@@ -17,3 +33,73 @@ def migrate(cr, version):
         "from account_voucher v "
         "join account_voucher_line l on l.voucher_id=v.id "
         "where v.tax_id is not null")
+
+def create_payments_from_vouchers(cr):
+    """Create account.payment rows from appropiate account.voucher rows."""
+    receipt_method = get_ref(
+        cr, 'account', 'account_payment_method_manual_in'
+    )
+    payment_method = get_ref(
+        cr, 'account', 'account_payment_method_manual_out'
+    )
+    # Keep original id, to later migrate references from other tables,
+    # although it seems not te be needed for now.
+    # Possible states of account.payment: draft, posted, sent, reconciled
+    # Possible states of account_voucher: draft, cancel, proforma, posted
+    # "type" field in 8.0 renamed to "voucher_type" during migration.
+    # Field destination_journal_id is used for internal transfer payments.
+    # Not used here.
+    # As far as I can see there is no connection between invoices or moves
+    # in 8.0 taht we can or have to migrate to 9.0.
+    cr.execute("""
+        INSERT INTO account_payment (
+            id, create_date, communication, company_id, currency_id,
+            partner_id, partner_type,
+            payment_method_id, payment_date, payment_difference_handling,
+            journal_id,
+            state, writeoff_account_id, create_uid, write_date, write_uid,
+            name, amount, payment_type, payment_reference
+        )
+        SELECT
+            v.id, v.create_date, v.comment, v.company_id,
+            v.payment_rate_currency_id, v.partner_id,
+            CASE
+               WHEN v.voucher_type = 'receipt' THEN 'customer'
+               ELSE 'supplier'
+            END,
+            CASE WHEN v.voucher_type = 'receipt' THEN %s ELSE %s END,
+            v.create_date,
+            CASE
+                WHEN v.payment_option = 'with_writeoff' THEN 'reconcile'
+                ELSE 'open'
+            END,
+            v.journal_id, v.state, v.writeoff_acc_id, v.create_uid,
+            v.write_date, v.write_uid,
+            COALESCE(v.name, v.comment),
+            v.amount,
+            CASE
+               WHEN v.voucher_type = 'receipt' THEN 'inbound'
+               ELSE 'outbound'
+            END,
+            v.reference
+        FROM account_voucher v
+        WHERE v.voucher_type IN ('receipt', 'payment')
+        AND v.state in ('draft', 'posted')""",
+        (receipt_method, payment_method)
+    )
+    cr.execute(
+        """SELECT COALESCE(MAX(id), 0) AS max_id FROM account_payment"""
+    )
+    max_id = cr.fetchone()[0] + 1
+    cr.execute(
+        """ALTER SEQUENCE account_payment_id_seq RESTART WITH %s""",
+        (max_id,)
+    )
+
+
+@openupgrade.migrate()
+def migrate(cr, version):
+    """Control function for account_voucher migration."""
+    create_payments_from_vouchers(cr)
+    migrate_voucher_types(cr)
+    create_voucher_line_tax_lines(cr)

--- a/addons/account_voucher/migrations/9.0.1.0/pre-migration.py
+++ b/addons/account_voucher/migrations/9.0.1.0/pre-migration.py
@@ -12,7 +12,8 @@ column_copies = {
 
 def delete_payment_views(cr):
     """Delete account.voucher payment views that hinder upgrade."""
-    cr.execute("""
+    cr.execute(
+        """\
         DELETE FROM ir_ui_view
         WHERE name like '%account.voucher.payment%'
         """

--- a/addons/account_voucher/migrations/9.0.1.0/pre-migration.py
+++ b/addons/account_voucher/migrations/9.0.1.0/pre-migration.py
@@ -10,8 +10,18 @@ column_copies = {
 }
 
 
+def delete_payment_views(cr):
+    """Delete account.voucher payment views that hinder upgrade."""
+    cr.execute("""
+        DELETE FROM ir_ui_view
+        WHERE name like '%account.voucher.payment%'
+        """
+    )
+
+
 @openupgrade.migrate()
 def migrate(cr, version):
     cr.execute('update account_voucher_line set amount=0 where amount is null')
     cr.execute("update account_voucher_line set name='/' where name is null")
     openupgrade.copy_columns(cr, column_copies)
+    delete_payment_views(cr)

